neo/rawio/spikegadgetsrawio.py

```diff
@@ -79,6 +79,26 @@ def __init__(self, filename="", selected_streams=None):
     def _source_name(self):
         return self.filename
 
+    def _produce_ephys_channel_ids(self, n_total_channels, n_channels_per_chip):
+        """Compute the channel ID labels
+        The ephys channels in the .rec file are stored in the following order:
+        hwChan ID of channel 0 of first chip, hwChan ID of channel 0 of second chip, ..., hwChan ID of channel 0 of Nth chip,
+        hwChan ID of channel 1 of first chip, hwChan ID of channel 1 of second chip, ..., hwChan ID of channel 1 of Nth chip,
+        ...
+        So if there are 32 channels per chip and 128 channels (4 chips), then the channel IDs are:
+        0, 32, 64, 96, 1, 33, 65, 97, ..., 128
+        See also: https://github.com/NeuralEnsemble/python-neo/issues/1215
+        """
+        x = []
+        for k in range(n_channels_per_chip):
+            x.append(
+                [
+                    k + i * n_channels_per_chip
+                    for i in range(int(n_total_channels / n_channels_per_chip))
+                ]
+            )
+        return [item for sublist in x for item in sublist]
```

Comment:
```suggestion
        channel_ids = [k + (i * n_channels_per_chip) for k in range(n_channels_per_chip) for i in range(int(n_total_channels / n_channels_per_chip)]

        return channel_ids
```

This is probably just me so if @alejoe91 takes a quick peek and is fine with the mixed for loop comprehension, but for me I prefer to keep it all list comprehension or nested for loops.  Mainly because I think flattening the nested lists is always super confusing for me. Let's see what Alessio thinks, but otherwise this looks fine to me.